bugsnag-spring/src/main/java/com/bugsnag/ExceptionClassCallback.java

```diff
@@ -0,0 +1,75 @@
+package com.bugsnag;
+
+import com.bugsnag.HandledState.SeverityReasonType;
+import com.bugsnag.callbacks.Callback;
+
+import org.springframework.beans.ConversionNotSupportedException;
+import org.springframework.beans.TypeMismatchException;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.http.converter.HttpMessageNotWritableException;
+import org.springframework.validation.BindException;
+import org.springframework.web.HttpMediaTypeNotAcceptableException;
+import org.springframework.web.HttpMediaTypeNotSupportedException;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.MissingPathVariableException;
+import org.springframework.web.bind.MissingServletRequestParameterException;
+import org.springframework.web.bind.ServletRequestBindingException;
+import org.springframework.web.multipart.support.MissingServletRequestPartException;
+import org.springframework.web.servlet.NoHandlerFoundException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A callback to assign a specified severity in a report for particular Spring exceptions.
+ * This severity can be overridden if the severity is set manually in an application.
+ */
+class ExceptionClassCallback implements Callback {
+
+    private static final Map<Class<? extends java.lang.Exception>, Severity> exceptionToSeverity;
+
+    static {
+        exceptionToSeverity = new HashMap<Class<? extends java.lang.Exception>, Severity>();
+        exceptionToSeverity.put(HttpRequestMethodNotSupportedException.class, Severity.INFO);
```

Comment:
Would be good to have comments for each of these explaining the reasoning behind the severity. At least for the non-error ones.